src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -824,14 +824,14 @@ internal void AddMethod(MethodInfo mi)
             return (start.StartLocation.Line, start.StartLocation.Column, end.EndLocation.Line, end.EndLocation.Column);
         }
 
-        private static async Task<MemoryStream> GetDataAsync(Uri uri, CancellationToken token)
+        private async Task<MemoryStream> GetDataAsync(Uri uri, CancellationToken token)
         {
             var mem = new MemoryStream();
             try
             {
                 if (uri.IsFile && File.Exists(uri.LocalPath))
                 {
-                    using (FileStream file = File.Open(uri.LocalPath, FileMode.Open))
+                    using (FileStream file = File.Open(SourceUri.LocalPath, FileMode.Open))
```

Comment:
This feels really fragile: `uri` is used to make a decision but then the load is being done from `SourceUri`.  There's one caller of this method and it does one call with `SourceUri` as the `uri` argument, and one with another Uri - it's going to be really painful to debug.

Can the logic inside `GetDataAsync` be made simpler (e.g. pass a flag to decide if it's a local file or an http request) and the `uri` analysis moved out to the caller?

Reply:
Oops, the change is a merge issue. This was recently fixed on main.
@thaystg